src/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD001UseSwitchToMainThreadAsyncAnalyzerTests.cs

```diff
@@ -107,12 +84,11 @@ class Test {
 }
 ";
 
-            this.expect.Locations = new[] { new DiagnosticResultLocation("Test0.cs", 7, 9, 7, 49) };
-            this.VerifyCSharpDiagnostic(test, this.expect);
+            await Verify.VerifyAnalyzerAsync(test, Verify.Diagnostic().WithSpan(7, 9, 7, 49));
```

Comment:
>Verify.Diagnostic().WithSpan(7, 9, 7, 49)) [](start = 51, length = 42)

So I'm curious how you decide when to use the hard-coded span approach vs. when to use the `[| |]` syntax.

Reply:
Markup syntax is optional.

Advantages: easy to see the span relative to code.

Disadvantages: no message validation, may be difficult to tell if a test has diagnostics or does not have diagnostics.

During the conversion process, I typically use markup syntax for code fix validation because most projects do not provide diagnostics for any code fix tests while the new library requires it, and markup is the least disruptive way to update the inputs to meet the new requirement.